crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -463,6 +464,19 @@ where
         .map(|expr| Statement::Constrain(ConstrainStatement(expr)))
 }
 
+fn early_return<'a, P>(expr_parser: P) -> impl NoirParser<Statement> + 'a
+where
+    P: ExprParser + 'a,
+{
+    ignore_then_commit(keyword(Keyword::Return), expr_parser)
+        .labelled("statement")
+        .map(|_| Statement::Error)
+        .validate(|expr, span, emit| {
+            emit(ParserError::with_reason(ParserErrorReason::EarlyReturnUnsupported, span));
+            expr
+        })
```

Comment:
`validate` also maps the input, so we can remove the map to return Statement::Error directly in validate